File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        
        while l <= r:
            m = l + (r - l) // 2
            mid = nums[m]
            
            if mid < target:
                l = m + 1
            elif mid > target:
                r = m - 1
            else: 
                return m
            
        return l

File: 0035-search-insert-position/0035-search-insert-position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut begin = 0;
        let mut end = nums.len();
        while begin != end {
            let mut current = (begin + end)/2;
            if nums[current] == target {
                return current as i32;
            }
            if nums[current] < target {
                begin = current + 1;
            }
            else {
                end = current;
            }
        }
        begin as i32
    }
}